Gateway/Request/CreditCardBuilder.php

```diff
@@ -21,15 +22,24 @@ class CreditCardBuilder implements BuilderInterface
     public function build(array $buildSubject)
     {
         $payment = SubjectReader::readPayment($buildSubject);
-        $method  = $payment->getPayment();
+        $method = $payment->getPayment();
+
+        $paymentData = null;
 
         if ($method->getAdditionalInformation(CreditCardDataObserver::CUSTOMER)) {
-            return [
+            $paymentData = [
                 self::CUSTOMER => $method->getAdditionalInformation(CreditCardDataObserver::CUSTOMER),
-                self::CARD     => $method->getAdditionalInformation(CreditCardDataObserver::CARD)
+                self::CARD => $method->getAdditionalInformation(CreditCardDataObserver::CARD),
+            ];
+        } else {
+            $paymentData = [
+                self::CARD => $method->getAdditionalInformation(CreditCardDataObserver::TOKEN)
             ];
         }
 
-        return [ self::CARD => $method->getAdditionalInformation(CreditCardDataObserver::TOKEN) ];
+        //add information about charge_id, if charge ID already exists than it is 'manual capture' request.
+        $paymentData[self::CHARGE_ID] = $method->getAdditionalInformation(CreditCardDataObserver::CHARGE_ID);
```

Comment:
Just continue from https://github.com/omise/omise-magento/pull/182/files#r250871230
This case is for `capture` only right? Do we need this to be assigned to `paymentData` variable (that contains `self::CARD` and `self::CUSTOMER`)?

Reply:
👆 Not sure if I can explain it properly but now the output from this method in the charge-capture case would be either
```php
$paymentData = [
    'self::CARD' = "",
    'self::CHARGE_ID' = "chrg_test_51qwe**",
]
```
or
```php
$paymentData = [
    'self::CUSTOMER' = "",
    'self::CARD' = "",
    'self::CHARGE_ID' = "chrg_test_51qwe**",
]
```
which we don't seem to need either `CARD` or `CUSTOMER` to be presented there in the array for the charge-capture case.

Reply:
this part was the same as it was... 
I just instead of doing instant return I did the first assign to a variable, then do something with that variable, and then return it. Before it was instant return.

Reply:
Sorry, I mean, do you need `self::CUSTOMER`, `self::CARD` to be presented in the array if it's just a case for charge-capture.
It was instant return for 2 cases, charge with a token, and charge with a saved-card from a specific Omise customer. But for the case of charge-capture, maybe we just need 
```php
return [ self::CHARGE_ID = $method->getAdditionalInformation(CreditCardDataObserver::CHARGE_ID) ];
```

Reply:
yes.. to manual capture we need only `charge_id`

Reply:
but I need to send in `paymentData` info about `charge_id` even if it is `null` all the time, because later there is decision made what that request is about based on `charge_id` status.